tv_apps/fling-player/test/unit/connector_test.js

```diff
@@ -0,0 +1,335 @@
+      msg = castingMsgTemplate.get().statusSeeked;
+      currSeq = connector.reportStatus(msg.status, { time : msg.time });
+      assertReportStatus(spy, msg, prevSeq, currSeq);
+    });
+
+    test('should report status of stopped', function () {
+      msg = castingMsgTemplate.get().statusStopped;
+      currSeq = connector.reportStatus(msg.status, { time : msg.time });
+      assertReportStatus(spy, msg, prevSeq, currSeq);
+    });
+
+    test('should report status of error', function () {
+      msg = castingMsgTemplate.get().statusError;
+      currSeq = connector.reportStatus(msg.status, {
+        time : msg.time,
+        error : msg.error
+      });
+      assertReportStatus(spy, msg, prevSeq, currSeq);
+    });
+
+    test('should not report status of unknown', function () {
+      msg = castingMsgTemplate.get().statusPlaying;
+      msg.status = 'unknown';
+      expect(() => connector.reportStatus(msg.status, { time : msg.time }))
+            .to.throw(/Ilegal status/);
+    });
+
+    test('should not report status without time', function () {
+      msg = castingMsgTemplate.get().statusPlaying;
+      expect(() => connector.reportStatus(msg.status, {}))
+            .to.throw(/Ilegal time/);
+    });
+  });
+
+  suite('handleRemoteMessage()', function () {
+
+    var spyOnLoadRequest, spyOnPlayRequest, spyOnSeekRequest, spyOnPauseRequest;
+
+    function assertHandled(spy, msg) {
+      assert.isTrue(
+        spy.withArgs(msg).calledOnce, 'Not hanlding remote message'
+      );
+    }
+
+    function assertNotHandled(spy, msg) {
+      assert.isFalse(
+        spy.withArgs(msg).calledOnce, 'Should not handle remote message'
+      );
+    }
+
+    function assertRequestCallCount(callCounts = {}) {
+      assert.equal(
+        spyOnLoadRequest.callCount, callCounts.loadRequestCount,
+        'Wrong load request count'
+      );
+      assert.equal(
+        spyOnPlayRequest.callCount, callCounts.playRequestCount,
+        'Wrong play request count'
+      );
+      assert.equal(
+        spyOnSeekRequest.callCount, callCounts.seekRequestCount,
+        'Wrong seek request count'
+      );
+      assert.equal(
+        spyOnPauseRequest.callCount, callCounts.pauseRequestCount,
+        'Wrong pause request count'
+      );
+    }
+
+    setup(function () {
+      spy = sinon.spy(connector, 'handleRemoteMessage');
+      spyOnLoadRequest = sinon.spy();
+      spyOnPlayRequest = sinon.spy();
+      spyOnSeekRequest = sinon.spy();
+      spyOnPauseRequest = sinon.spy();
+      connector.on('loadRequest', spyOnLoadRequest);
+      connector.on('playRequest', spyOnPlayRequest);
+      connector.on('seekRequest', spyOnSeekRequest);
+      connector.on('pauseRequest', spyOnPauseRequest);
+    });
+
+    test('should fire loadRequest on load message', function () {
+      msg = castingMsgTemplate.get().load;
+      presentation.mCastMsgToReceiver(msg);
+      assertHandled(spy, msg);
```

Comment:
So what you do here is actually making asserting for `onConnectionMessage()` and then asserting for `handleRemoteMessage`, which is not easily to understand for categorized under `handleRemoteMessage()` test cases.